sonarr/qualitydefinition.go

```diff
@@ -0,0 +1,62 @@
+package sonarr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
+// GetQualityDefinitions returns all configured quality definitions.
+func (s *Sonarr) GetQualityDefinitions() ([]*QualityDefinition, error) {
+	return s.GetQualityDefinitionsContext(context.Background())
+}
+
+func (s *Sonarr) GetQualityDefinitionsContext(ctx context.Context) ([]*QualityDefinition, error) {
+	var definitions []*QualityDefinition
+
+	if _, err := s.GetInto(ctx, "v3/qualityDefinition", nil, &definitions); err != nil {
+		return nil, fmt.Errorf("api.Get(qualityDefinition): %w", err)
+	}
+
+	return definitions, nil
+}
+
+// GetQualityDefinition returns a single quality definition.
+func (s *Sonarr) GetQualityDefinition(qualityDefinitionID int) (*QualityDefinition, error) {
+	return s.GetQualityDefinitionContext(context.Background(), qualityDefinitionID)
+}
+
+func (s *Sonarr) GetQualityDefinitionContext(ctx context.Context, qualityDefinitionID int) (*QualityDefinition, error) {
+	var definition *QualityDefinition
+
+	if _, err := s.GetInto(ctx, "v3/qualityDefinition/"+strconv.Itoa(qualityDefinitionID), nil, &definition); err != nil {
+		return nil, fmt.Errorf("api.Get(qualityDefinition): %w", err)
+	}
+
+	return definition, nil
+}
+
+// UpdateQualityDefinition updates the quality definition.
+func (s *Sonarr) UpdateQualityDefinition(definition *QualityDefinition) (*QualityDefinition, error) {
+	return s.UpdateQualityDefinitionContext(context.Background(), definition)
+}
+
+func (s *Sonarr) UpdateQualityDefinitionContext(ctx context.Context,
+	definition *QualityDefinition) (*QualityDefinition, error) {
+	var (
+		body   bytes.Buffer
+		output QualityDefinition
+	)
+
+	if err := json.NewEncoder(&body).Encode(definition); err != nil {
+		return nil, fmt.Errorf("json.Marshal(qualityDefinition): %w", err)
+	}
+
+	if _, err := s.PutInto(ctx, "v3/qualityDefinition/"+strconv.Itoa(int(definition.ID)), nil, &body, &output); err != nil {
```

Comment:
Generally, and especially with these _small_ methods, I like putting the assignment in the if statement, but sometimes it makes the lines a little longer than I like. In those cases, I put the assignment and conditional on different lines. Basically, if the linter yells `lll` I cut them up. You can run the linter locally with `make test`. Just make sure to install the correct version of golangci-lint (see .travis.yml).

```
GOOS=linux golangci-lint run --enable-all -D maligned,scopelint,interfacer,golint,tagliatelle,exhaustivestruct,dupl
sonarr/qualitydefinition.go:57: line is 121 characters (lll)
Makefile:11: recipe for target 'lint' failed
```

```suggestion
	_, err := s.PutInto(ctx, "v3/qualityDefinition/"+strconv.Itoa(int(definition.ID)), nil, &body, &output);
	if err != nil {
```

Reply:
I split up the line without making another command. I think it's better to limit the scope of "err" whenever we can. what do you think?

Reply:
Also, sorry for the linter, I'll keep that in mind!